framework/src/meshgenerators/MeshRepairGenerator.C

```diff
@@ -61,112 +63,136 @@ MeshRepairGenerator::generate()
+{
+  unsigned int num_fixed_nodes = 0;
+  auto pl = mesh->sub_point_locator();
+  pl->set_close_to_point_tol(_node_overlap_tol);
+
+  std::unordered_set<dof_id_type> nodes_removed;
+  // loop on nodes
+  for (auto & node : mesh->node_ptr_range())
   {
-    _num_fixed_nodes = 0;
-    auto pl = mesh->sub_point_locator();
-    std::unordered_set<dof_id_type> nodes_removed;
-    // loop on nodes
-    for (auto & node : mesh->node_ptr_range())
-    {
-      // this node has already been removed
-      if (nodes_removed.count(node->id()))
-        continue;
+    // this node has already been removed
+    if (nodes_removed.count(node->id()))
+      continue;
 
-      // find all the elements around this node
-      std::set<const Elem *> elements;
-      (*pl)(*node, elements);
+    // find all the elements around this node
+    std::set<const Elem *> elements;
+    (*pl)(*node, elements);
 
-      for (auto & elem : elements)
+    for (auto & elem : elements)
+    {
+      bool found = false;
+      for (auto & elem_node : elem->node_ref_range())
       {
-        bool found = false;
-        for (auto & elem_node : elem->node_ref_range())
+        if (node->id() == elem_node.id())
         {
-          if (node->id() == elem_node.id())
-          {
-            found = true;
-            break;
-          }
+          found = true;
+          break;
         }
-        if (!found)
+      }
+      if (!found)
+      {
+        for (auto & elem_node : elem->node_ref_range())
         {
-          for (auto & elem_node : elem->node_ref_range())
+          if (elem_node.id() == node->id())
+            continue;
+          const Real tol = _node_overlap_tol;
+          // Compares the coordinates
+          const auto x_node = (*node)(0);
+          const auto x_elem_node = elem_node(0);
+          const auto y_node = (*node)(1);
+          const auto y_elem_node = elem_node(1);
+          const auto z_node = (*node)(2);
+          const auto z_elem_node = elem_node(2);
+
+          if (MooseUtils::absoluteFuzzyEqual(x_node, x_elem_node, tol) &&
+              MooseUtils::absoluteFuzzyEqual(y_node, y_elem_node, tol) &&
+              MooseUtils::absoluteFuzzyEqual(z_node, z_elem_node, tol))
           {
-            const Real tol = _node_overlap_tol;
-            // Compares the coordinates
-            const auto x_node = (*node)(0);
-            const auto x_elem_node = elem_node(0);
-            const auto y_node = (*node)(1);
-            const auto y_elem_node = elem_node(1);
-            const auto z_node = (*node)(2);
-            const auto z_elem_node = elem_node(2);
-
-            if (MooseUtils::absoluteFuzzyEqual(x_node, x_elem_node, tol) &&
-                MooseUtils::absoluteFuzzyEqual(y_node, y_elem_node, tol) &&
-                MooseUtils::absoluteFuzzyEqual(z_node, z_elem_node, tol))
+            // Merging two nodes from the same element is almost never a good idea
```

Comment:
Good catch.  You'd think nobody would ever do this on purpose, but it's apparently quite popular for some codes to simulate a prism by collapsing one Hex face into an edge and then adding some constraints to prevent a discontinuity there.